tests/unit/test_main.py

```diff
@@ -0,0 +1,104 @@
+"""
+Test config
+
+Copyright (c) 2024 Aiven Ltd
+See LICENSE for details
+"""
+
+import runpy
+from unittest.mock import patch, MagicMock
+
+from src.karapace import __main__
```

Comment:
This is highly problematic. The `src` directory structure has the exact purpose of never executing code by local path in tests, but to enforce always executing what is _installed_. In other words, imports should never be prefixed with `src.*`.

Reply:
See e.g. https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/